src/java/org/tensorics/core/tensor/AbstractTensorBuilder.java

```diff
@@ -100,18 +89,35 @@ private void checkIfContextValid(Position context2) {
 		}
 	}
 
-	
+	@Override
+	public void putAll(Tensor<E> tensor) {
+		this.putAll(Position.empty(), tensor);
+	}
 
 	@Override
 	public final void putAll(Position position, Tensor<E> tensor) {
 		checkNotNull(tensor, "The tensor must not be null!");
+		putAll(position, TensorInternals.mapFrom(tensor));
+	}
+
+	@Override
+	public void putAll(Map<Position, E> newEntries) {
+		putAll(Position.empty(), newEntries);
+	}
+
+	@Override

```

Comment:
As discussed today, I think I would leave it for the moment, since it seems to be useful in certain situations, where a tensor cannot easily be constructed, but a map is available. (E.g. internally)
And at least the API is symmetric